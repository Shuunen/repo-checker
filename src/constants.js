import { join } from 'path'

export const home = process.env.HOME
export const dataFileName = 'repo-checker.config.js'
export const repoCheckerPath = join(__dirname, '..')
export const templatePath = join(repoCheckerPath, 'templates')
export const dataFileHomePath = join(home, dataFileName)

export const dataDefaults = {
  auto_merge: true,
  ban_sass: true,
  dev_deps_only: true,
  license: 'GPL-3.0',
  max_size_ko: 10,
  user_mail: 'romain.racamier@gmail.com',
  user_name: 'Romain Racamier-Lafon',
}
