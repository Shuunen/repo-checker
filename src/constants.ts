import { join } from 'path'

/* istanbul ignore next */
export const home = process.env.HOME ?? '' // when does HOME is not defined ?
export const dataFileName = 'repo-checker.config.js'
/* istanbul ignore next */
export const repoCheckerPath = process.env.pwd ?? process.cwd()
export const templatePath = join(repoCheckerPath, 'templates')
export const dataFileHomePath = join(home, dataFileName)

export class ProjectData {
  auto_merge = true
  ban_sass = true
  dev_deps_only = true
  is_module = false
  license = 'GPL-3.0'
  max_size_ko = 1
  npm_package = false
  package_name = ''
  quiet = false
  repo_id = 'a-great-repo'
  use_typescript = false
  use_vue = false
  user_id = 'Shuunen'
  user_id_lowercase = 'shuunen'
  user_mail = 'romain.racamier@gmail.com'
  user_name = 'Romain Racamier-Lafon'
  web_published = false
  web_url = 'https://my-website.com'

  constructor (data: Partial<ProjectData> = {}) {
    Object.assign(this, data)
  }
}

export const dataDefaults = new ProjectData()
