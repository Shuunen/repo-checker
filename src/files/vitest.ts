import { FileBase } from '../file.ts'

// eslint-disable-next-line no-export
export class VitestFile extends FileBase {
  // eslint-disable-next-line jsdoc/require-jsdoc
  // eslint-disable-next-line max-lines-per-function
  public async start() {
    if (!this.data.isUsingVitest) return
    const hasMtsFile = await this.fileExists('vitest.config.mts')
    const hasJsFile = await this.fileExists('vitest.config.js')
    const hasTsFile = await this.fileExists('vitest.config.ts')
    const hasMjsFile = await this.fileExists('vitest.config.mjs')
    if (hasMtsFile) await this.inspectFile('vitest.config.mts')
    /* c8 ignore next 3 */
    if (hasJsFile) await this.inspectFile('vitest.config.js')
    if (hasTsFile) await this.inspectFile('vitest.config.ts')
    if (hasMjsFile) await this.inspectFile('vitest.config.mjs')
    this.couldContains('a pool thread setting', /pool: 'threads'/u)
  }
}
