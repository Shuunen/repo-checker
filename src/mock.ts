/* eslint-disable jsdoc/require-jsdoc */
import path from 'node:path'
import { clone, sleep } from 'shuutils'
import type { Indicators } from './check.ts'
import type { FileBase } from './file.ts'
import { join } from './utils.ts'

const fileBaseKeysToDelete: (keyof FileBase)[] = ['fileExists', 'checkFileExists', 'folderPath', 'inspectFile', 'updateFile', 'initFile']
const indicatorsKeys: ReadonlyArray<keyof Indicators> = ['failed', 'passed', 'warnings'] as const

function cleanStringForSnap(input: string) {
  return (
    input
      // replace repo-check-1-40 by repo-check-x-yy
      .replace(/repo-check-\d-\d+/gu, 'repo-check-x-yy')
      // replace package-json-main-file-size-63ko by package-json-main-file-size-XYZko
      .replace(/package-json-main-file-size-\d+ko/gu, 'package-json-main-file-size-XYZko')
  )
}

// eslint-disable-next-line no-restricted-syntax
export function cleanUnknownValueForSnap<Type>(input: Type): Type {
  /* c8 ignore next 8 */
  /* eslint-disable @typescript-eslint/consistent-type-assertions, @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-type-assertion */
  if (typeof input === 'string') return cleanStringForSnap(input) as Type
  if (Array.isArray(input)) return input.map(item => cleanUnknownValueForSnap(item)) as Type
  if (typeof input === 'object') return JSON.parse(cleanStringForSnap(JSON.stringify(input))) as Type
  // eslint-disable-next-line no-restricted-syntax
  throw new Error(`cleanInstanceValueForSnap: unknown type ${typeof input}`)
  /* eslint-enable @typescript-eslint/consistent-type-assertions, @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-type-assertion */
}

export async function promiseValue<Type>(value: Type) {
  await sleep(1)
  return value
}

export async function promiseTrue() {
  await sleep(1)
  return true
}

export async function promiseFalse() {
  await sleep(1)
  return false
}

export async function promiseVoid() {
  await sleep(1)
}

// eslint-disable-next-line unicorn/prefer-module
export const sourceFolder = path.dirname(__filename)

export const vueProjectFolder = join(sourceFolder, 'mocks', 'vueProject')

export const tsProjectFolder = join(sourceFolder, 'mocks', 'tsProject')

export const mocksProjectsFolder = join(sourceFolder, 'mocks')

export function cleanIndicatorsForSnap(record?: Readonly<Record<string, unknown>>) {
  if (!record) return record
  const clean = clone(record)
  for (const key of indicatorsKeys) if (key in clean) clean[key] = cleanUnknownValueForSnap(clean[key])
  return clean
}

// eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
export function cleanInstanceForSnap(instance: Readonly<FileBase>, ...bonusKeys: readonly string[]) {
  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
  const clean = clone<Partial<FileBase>>(instance) as Record<string, unknown>
  const keys = [...fileBaseKeysToDelete, ...bonusKeys] // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
  for (const key of keys) if (key in clean) delete clean[key]
  return cleanIndicatorsForSnap(clean)
}
