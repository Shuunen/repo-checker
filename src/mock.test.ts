import { mkdirSync, rmSync } from 'node:fs'
import { Result } from 'shuutils'
import { expect, it } from 'vitest'
import { dataDefaults, dataFileName, repoCheckerPath } from './constants'
// eslint-disable-next-line max-dependencies
import { promiseValue, sourceFolder, tsProjectFolder, vueProjectFolder } from './mock'
import { augmentData, augmentDataWithGit, augmentDataWithPackageJson, findInFolder, getFileSizeInKo, getProjectFolders, isProjectFolder, join, jsToJson, messageToCode, objectToJson, readFileInFolder, writeFile } from './utils'

it('isProjectFolder A', async () => {
  expect(await isProjectFolder(repoCheckerPath)).toBe(true)
})

it('isProjectFolder B folders listing', async () => {
  expect(await getProjectFolders(repoCheckerPath)).toStrictEqual([repoCheckerPath])
  const projects = ['anotherProject', 'sampleProject']
  for (const name of projects) {
    const folderPath = join(sourceFolder, name, '.git')
    // eslint-disable-next-line @typescript-eslint/naming-convention
    mkdirSync(folderPath, { recursive: true })
    // eslint-disable-next-line no-await-in-loop
    await writeFile(join(folderPath, 'config'), '', 'utf8')
  }
  const folders = await getProjectFolders(sourceFolder)
  expect(folders.length >= 2).toBe(true)
  // eslint-disable-next-line @typescript-eslint/naming-convention
  for (const name of projects) rmSync(join(sourceFolder, name), { recursive: true })
})

it('readFileInFolder A read folder instead of file', async () => {
  const result = Result.unwrap(await readFileInFolder(sourceFolder, ''))
  expect(result.value).toBeUndefined()
  expect(result.error).toContain('is a directory')
})

const filename = 'test-file.log'

it('readFileInFolder B file does not exists', async () => {
  const result = Result.unwrap(await readFileInFolder('/', filename))
  expect(result.value).toBeUndefined()
  expect(result.error).toContain('does not exists')
})

it('file size calculation', async () => {
  const nonExistingFileSize = await getFileSizeInKo(filename)
  expect(nonExistingFileSize).toBe(0)
  const existingFileSize = await getFileSizeInKo('package.json')
  expect(existingFileSize >= 1).toBe(true)
})

it('augmentData A repoCheckerPath', async () => {
  expect(await augmentData(repoCheckerPath, dataDefaults)).toMatchSnapshot()
})
it('augmentData B vueProjectFolder', async () => {
  expect(await augmentData(vueProjectFolder, dataDefaults)).toMatchSnapshot()
})
it('augmentData C tsProjectFolder', async () => {
  expect(await augmentData(tsProjectFolder, dataDefaults)).toMatchSnapshot()
})
it('augmentData D testFolder', async () => {
  expect(await augmentData(sourceFolder, dataDefaults)).toMatchSnapshot()
})
it('augmentData A repoCheckerPath with local', async () => {
  expect(await augmentData(repoCheckerPath, dataDefaults, true)).toMatchSnapshot()
})
it('augmentData B vueProjectFolder with local', async () => {
  expect(await augmentData(vueProjectFolder, dataDefaults, true)).toMatchSnapshot()
})
it('augmentData C tsProjectFolder with local', async () => {
  expect(await augmentData(tsProjectFolder, dataDefaults, true)).toMatchSnapshot()
})
it('augmentData D testFolder with local', async () => {
  expect(await augmentData(sourceFolder, dataDefaults, true)).toMatchSnapshot()
})

it('augmentDataWithGit A repoCheckerPath', async () => {
  expect(await augmentDataWithGit(repoCheckerPath, dataDefaults)).toMatchSnapshot()
})
it('augmentDataWithGit B vueProjectFolder', async () => {
  expect(await augmentDataWithGit(vueProjectFolder, dataDefaults)).toMatchSnapshot()
})
it('augmentDataWithGit C tsProjectFolder', async () => {
  expect(await augmentDataWithGit(tsProjectFolder, dataDefaults)).toMatchSnapshot()
})
it('augmentDataWithGit D testFolder', async () => {
  expect(await augmentDataWithGit(sourceFolder, dataDefaults)).toMatchSnapshot()
})
it('augmentDataWithGit E non-existing folder', async () => {
  expect(await augmentDataWithGit('/non-existing-folder', dataDefaults)).toMatchSnapshot()
})

it('augmentDataWithPackageJson A repoCheckerPath', async () => {
  expect(await augmentDataWithPackageJson(repoCheckerPath, dataDefaults)).toMatchSnapshot()
})
it('augmentDataWithPackageJson B vueProjectFolder', async () => {
  expect(await augmentDataWithPackageJson(vueProjectFolder, dataDefaults)).toMatchSnapshot()
})
it('augmentDataWithPackageJson C tsProjectFolder', async () => {
  expect(await augmentDataWithPackageJson(tsProjectFolder, dataDefaults)).toMatchSnapshot()
})
it('augmentDataWithPackageJson D testFolder', async () => {
  expect(await augmentDataWithPackageJson(sourceFolder, dataDefaults)).toMatchSnapshot()
})

it('find pattern in folder', async () => {
  const result = await findInFolder(tsProjectFolder, /Dwight Schrute/u)
  expect(result[0]).toBe(dataFileName)
})

it('find pattern in sub folder', async () => {
  const result = await findInFolder(tsProjectFolder, /Alice/u)
  expect(result[0]).toBe('here.txt')
})

it('find no pattern in folder', async () => {
  const result = await findInFolder(tsProjectFolder, /Bob/u)
  expect(result.length).toBe(0)
})

it('find is skipped when scanning node_modules or git folders', async () => {
  const result = await findInFolder(repoCheckerPath, /blob volley/u)
  expect(result).toStrictEqual(['mock.test.ts'])
})

it('message to code', () => {
  expect(messageToCode('hello world')).toBe('hello-world')
  expect(messageToCode('hello/world::!')).toBe('hello-world')
  expect(messageToCode('Hey this is :)the _Hello/world::!')).toBe('hey-this-is-the-hello-world')
})

it('jsToJson A', () => {
  expect(jsToJson('a')).toBe('a')
})

it('jsToJson B', () => {
  expect(
    jsToJson(`/* some comment */
module.exports = {
  userName: 'Dwight Schrute',
  banSass: false,
}`),
  ).toBe(`{
  "userName": "Dwight Schrute",
  "banSass": false
}`)
})

it('objectToJson A nothing to sort', () => {
  expect(objectToJson({ keyA: 'A' })).toMatchSnapshot()
})

it('objectToJson B already sorted', () => {
  expect(objectToJson({ keyA: 'A', keyB: 'B' })).toMatchSnapshot()
})

it('objectToJson C sort', () => {
  expect(objectToJson({ keyA: 'A', keyB: 'B' })).toMatchSnapshot()
})

it('promiseValue A', async () => {
  expect(await promiseValue('a')).toBe('a')
})
