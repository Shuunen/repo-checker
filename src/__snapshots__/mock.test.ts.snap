// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`augmentData A repoCheckerPath 1`] = `
ProjectData {
  "canAutoMergeDeps": true,
  "hasMainBranch": false,
  "hasTaskPrefix": true,
  "isModule": false,
  "isPublishedPackage": true,
  "isQuiet": false,
  "isUsingC8": false,
  "isUsingDependencyCruiser": true,
  "isUsingEslint": true,
  "isUsingNyc": false,
  "isUsingShuutils": true,
  "isUsingTailwind": false,
  "isUsingTypescript": true,
  "isUsingV8": true,
  "isUsingVue": false,
  "isWebPublished": false,
  "license": "GPL-3.0",
  "maxSizeKo": 1,
  "packageName": "repo-check",
  "repoId": "repo-checker",
  "shouldAvoidSass": true,
  "userId": "Shuunen",
  "userIdLowercase": "shuunen",
  "userMail": "romain.racamier@gmail.com",
  "userName": "Romain Racamier-Lafon",
  "webUrl": "https://my-website.com",
  "willGenerateReport": true,
}
`;

exports[`augmentData A repoCheckerPath with local 1`] = `
ProjectData {
  "canAutoMergeDeps": true,
  "hasMainBranch": false,
  "hasTaskPrefix": true,
  "isModule": false,
  "isPublishedPackage": true,
  "isQuiet": false,
  "isUsingC8": false,
  "isUsingDependencyCruiser": true,
  "isUsingEslint": true,
  "isUsingNyc": false,
  "isUsingShuutils": true,
  "isUsingTailwind": false,
  "isUsingTypescript": true,
  "isUsingV8": true,
  "isUsingVue": false,
  "isWebPublished": false,
  "license": "GPL-3.0",
  "maxSizeKo": 70,
  "packageName": "repo-check",
  "repoId": "repo-checker",
  "shouldAvoidSass": true,
  "userId": "Shuunen",
  "userIdLowercase": "shuunen",
  "userMail": "romain.racamier@gmail.com",
  "userName": "Romain Racamier-Lafon",
  "webUrl": "https://my-website.com",
  "willGenerateReport": true,
}
`;

exports[`augmentData B vueProjectFolder 1`] = `
ProjectData {
  "canAutoMergeDeps": true,
  "hasMainBranch": false,
  "hasTaskPrefix": false,
  "isModule": false,
  "isPublishedPackage": false,
  "isQuiet": false,
  "isUsingC8": false,
  "isUsingDependencyCruiser": false,
  "isUsingEslint": false,
  "isUsingNyc": false,
  "isUsingShuutils": false,
  "isUsingTailwind": true,
  "isUsingTypescript": false,
  "isUsingV8": false,
  "isUsingVue": true,
  "isWebPublished": true,
  "license": "GPL-3.0",
  "maxSizeKo": 1,
  "packageName": "name",
  "repoId": "a-great-repo",
  "shouldAvoidSass": true,
  "userId": "Kevin_Malone",
  "userIdLowercase": "kevin_malone",
  "userMail": "romain.racamier@gmail.com",
  "userName": "Romain Racamier-Lafon",
  "webUrl": "https://my-website.com",
  "willGenerateReport": true,
}
`;

exports[`augmentData B vueProjectFolder with local 1`] = `
ProjectData {
  "canAutoMergeDeps": true,
  "hasMainBranch": false,
  "hasTaskPrefix": false,
  "isModule": false,
  "isPublishedPackage": false,
  "isQuiet": false,
  "isUsingC8": false,
  "isUsingDependencyCruiser": false,
  "isUsingEslint": false,
  "isUsingNyc": false,
  "isUsingShuutils": false,
  "isUsingTailwind": true,
  "isUsingTypescript": false,
  "isUsingV8": false,
  "isUsingVue": true,
  "isWebPublished": true,
  "license": "GPL-3.0",
  "maxSizeKo": 1,
  "packageName": "name",
  "repoId": "a-great-repo",
  "shouldAvoidSass": true,
  "userId": "Kevin_Malone",
  "userIdLowercase": "kevin_malone",
  "userMail": "romain.racamier@gmail.com",
  "userName": "Romain Racamier-Lafon",
  "webUrl": "https://my-website.com",
  "willGenerateReport": true,
}
`;

exports[`augmentData C tsProjectFolder 1`] = `
ProjectData {
  "canAutoMergeDeps": true,
  "hasMainBranch": false,
  "hasTaskPrefix": true,
  "isModule": false,
  "isPublishedPackage": false,
  "isQuiet": false,
  "isUsingC8": false,
  "isUsingDependencyCruiser": false,
  "isUsingEslint": false,
  "isUsingNyc": false,
  "isUsingShuutils": false,
  "isUsingTailwind": false,
  "isUsingTypescript": true,
  "isUsingV8": false,
  "isUsingVue": false,
  "isWebPublished": false,
  "license": "MIT",
  "maxSizeKo": 1,
  "packageName": "",
  "repoId": "a-great-repo",
  "shouldAvoidSass": true,
  "userId": "Shuunen",
  "userIdLowercase": "shuunen",
  "userMail": "romain.racamier@gmail.com",
  "userName": "Michael Scott-Junior",
  "webUrl": "https://my-website.com",
  "willGenerateReport": true,
}
`;

exports[`augmentData C tsProjectFolder with local 1`] = `
ProjectData {
  "canAutoMergeDeps": true,
  "hasMainBranch": false,
  "hasTaskPrefix": true,
  "isModule": false,
  "isPublishedPackage": false,
  "isQuiet": false,
  "isUsingC8": false,
  "isUsingDependencyCruiser": false,
  "isUsingEslint": false,
  "isUsingNyc": false,
  "isUsingShuutils": false,
  "isUsingTailwind": false,
  "isUsingTypescript": true,
  "isUsingV8": false,
  "isUsingVue": false,
  "isWebPublished": false,
  "license": "MIT",
  "maxSizeKo": 1,
  "packageName": "",
  "repoId": "a-great-repo",
  "shouldAvoidSass": false,
  "userId": "Shuunen",
  "userIdLowercase": "shuunen",
  "userMail": "romain.racamier@gmail.com",
  "userName": "Dwight Schrute",
  "webUrl": "https://my-website.com",
  "willGenerateReport": true,
}
`;

exports[`augmentData D testFolder 1`] = `
ProjectData {
  "canAutoMergeDeps": true,
  "hasMainBranch": false,
  "hasTaskPrefix": false,
  "isModule": false,
  "isPublishedPackage": false,
  "isQuiet": false,
  "isUsingC8": false,
  "isUsingDependencyCruiser": false,
  "isUsingEslint": false,
  "isUsingNyc": false,
  "isUsingShuutils": false,
  "isUsingTailwind": false,
  "isUsingTypescript": false,
  "isUsingV8": false,
  "isUsingVue": false,
  "isWebPublished": false,
  "license": "GPL-3.0",
  "maxSizeKo": 1,
  "packageName": "",
  "repoId": "a-great-repo",
  "shouldAvoidSass": true,
  "userId": "Shuunen",
  "userIdLowercase": "shuunen",
  "userMail": "romain.racamier@gmail.com",
  "userName": "Romain Racamier-Lafon",
  "webUrl": "https://my-website.com",
  "willGenerateReport": true,
}
`;

exports[`augmentData D testFolder with local 1`] = `
ProjectData {
  "canAutoMergeDeps": true,
  "hasMainBranch": false,
  "hasTaskPrefix": false,
  "isModule": false,
  "isPublishedPackage": false,
  "isQuiet": false,
  "isUsingC8": false,
  "isUsingDependencyCruiser": false,
  "isUsingEslint": false,
  "isUsingNyc": false,
  "isUsingShuutils": false,
  "isUsingTailwind": false,
  "isUsingTypescript": false,
  "isUsingV8": false,
  "isUsingVue": false,
  "isWebPublished": false,
  "license": "GPL-3.0",
  "maxSizeKo": 1,
  "packageName": "",
  "repoId": "a-great-repo",
  "shouldAvoidSass": true,
  "userId": "Shuunen",
  "userIdLowercase": "shuunen",
  "userMail": "romain.racamier@gmail.com",
  "userName": "Romain Racamier-Lafon",
  "webUrl": "https://my-website.com",
  "willGenerateReport": true,
}
`;

exports[`augmentDataWithGit A repoCheckerPath 1`] = `
ProjectData {
  "canAutoMergeDeps": true,
  "hasMainBranch": false,
  "hasTaskPrefix": false,
  "isModule": false,
  "isPublishedPackage": false,
  "isQuiet": false,
  "isUsingC8": false,
  "isUsingDependencyCruiser": false,
  "isUsingEslint": false,
  "isUsingNyc": false,
  "isUsingShuutils": false,
  "isUsingTailwind": false,
  "isUsingTypescript": false,
  "isUsingV8": false,
  "isUsingVue": false,
  "isWebPublished": false,
  "license": "GPL-3.0",
  "maxSizeKo": 1,
  "packageName": "",
  "repoId": "repo-checker",
  "shouldAvoidSass": true,
  "userId": "Shuunen",
  "userIdLowercase": "shuunen",
  "userMail": "romain.racamier@gmail.com",
  "userName": "Romain Racamier-Lafon",
  "webUrl": "https://my-website.com",
  "willGenerateReport": true,
}
`;

exports[`augmentDataWithGit B vueProjectFolder 1`] = `
ProjectData {
  "canAutoMergeDeps": true,
  "hasMainBranch": false,
  "hasTaskPrefix": false,
  "isModule": false,
  "isPublishedPackage": false,
  "isQuiet": false,
  "isUsingC8": false,
  "isUsingDependencyCruiser": false,
  "isUsingEslint": false,
  "isUsingNyc": false,
  "isUsingShuutils": false,
  "isUsingTailwind": false,
  "isUsingTypescript": false,
  "isUsingV8": false,
  "isUsingVue": false,
  "isWebPublished": false,
  "license": "GPL-3.0",
  "maxSizeKo": 1,
  "packageName": "",
  "repoId": "a-great-repo",
  "shouldAvoidSass": true,
  "userId": "Shuunen",
  "userIdLowercase": "shuunen",
  "userMail": "romain.racamier@gmail.com",
  "userName": "Romain Racamier-Lafon",
  "webUrl": "https://my-website.com",
  "willGenerateReport": true,
}
`;

exports[`augmentDataWithGit C tsProjectFolder 1`] = `
ProjectData {
  "canAutoMergeDeps": true,
  "hasMainBranch": false,
  "hasTaskPrefix": false,
  "isModule": false,
  "isPublishedPackage": false,
  "isQuiet": false,
  "isUsingC8": false,
  "isUsingDependencyCruiser": false,
  "isUsingEslint": false,
  "isUsingNyc": false,
  "isUsingShuutils": false,
  "isUsingTailwind": false,
  "isUsingTypescript": false,
  "isUsingV8": false,
  "isUsingVue": false,
  "isWebPublished": false,
  "license": "GPL-3.0",
  "maxSizeKo": 1,
  "packageName": "",
  "repoId": "a-great-repo",
  "shouldAvoidSass": true,
  "userId": "Shuunen",
  "userIdLowercase": "shuunen",
  "userMail": "romain.racamier@gmail.com",
  "userName": "Romain Racamier-Lafon",
  "webUrl": "https://my-website.com",
  "willGenerateReport": true,
}
`;

exports[`augmentDataWithGit D testFolder 1`] = `
ProjectData {
  "canAutoMergeDeps": true,
  "hasMainBranch": false,
  "hasTaskPrefix": false,
  "isModule": false,
  "isPublishedPackage": false,
  "isQuiet": false,
  "isUsingC8": false,
  "isUsingDependencyCruiser": false,
  "isUsingEslint": false,
  "isUsingNyc": false,
  "isUsingShuutils": false,
  "isUsingTailwind": false,
  "isUsingTypescript": false,
  "isUsingV8": false,
  "isUsingVue": false,
  "isWebPublished": false,
  "license": "GPL-3.0",
  "maxSizeKo": 1,
  "packageName": "",
  "repoId": "a-great-repo",
  "shouldAvoidSass": true,
  "userId": "Shuunen",
  "userIdLowercase": "shuunen",
  "userMail": "romain.racamier@gmail.com",
  "userName": "Romain Racamier-Lafon",
  "webUrl": "https://my-website.com",
  "willGenerateReport": true,
}
`;

exports[`augmentDataWithPackageJson A repoCheckerPath 1`] = `
ProjectData {
  "canAutoMergeDeps": true,
  "hasMainBranch": false,
  "hasTaskPrefix": true,
  "isModule": false,
  "isPublishedPackage": true,
  "isQuiet": false,
  "isUsingC8": false,
  "isUsingDependencyCruiser": true,
  "isUsingEslint": true,
  "isUsingNyc": false,
  "isUsingShuutils": true,
  "isUsingTailwind": false,
  "isUsingTypescript": true,
  "isUsingV8": true,
  "isUsingVue": false,
  "isWebPublished": false,
  "license": "GPL-3.0",
  "maxSizeKo": 1,
  "packageName": "repo-check",
  "repoId": "a-great-repo",
  "shouldAvoidSass": true,
  "userId": "Shuunen",
  "userIdLowercase": "shuunen",
  "userMail": "romain.racamier@gmail.com",
  "userName": "Romain Racamier-Lafon",
  "webUrl": "https://my-website.com",
  "willGenerateReport": true,
}
`;

exports[`augmentDataWithPackageJson B vueProjectFolder 1`] = `
ProjectData {
  "canAutoMergeDeps": true,
  "hasMainBranch": false,
  "hasTaskPrefix": false,
  "isModule": false,
  "isPublishedPackage": false,
  "isQuiet": false,
  "isUsingC8": false,
  "isUsingDependencyCruiser": false,
  "isUsingEslint": false,
  "isUsingNyc": false,
  "isUsingShuutils": false,
  "isUsingTailwind": true,
  "isUsingTypescript": false,
  "isUsingV8": false,
  "isUsingVue": true,
  "isWebPublished": true,
  "license": "GPL-3.0",
  "maxSizeKo": 1,
  "packageName": "name",
  "repoId": "a-great-repo",
  "shouldAvoidSass": true,
  "userId": "Kevin_Malone",
  "userIdLowercase": "kevin_malone",
  "userMail": "romain.racamier@gmail.com",
  "userName": "Romain Racamier-Lafon",
  "webUrl": "https://my-website.com",
  "willGenerateReport": true,
}
`;

exports[`augmentDataWithPackageJson C tsProjectFolder 1`] = `
ProjectData {
  "canAutoMergeDeps": true,
  "hasMainBranch": false,
  "hasTaskPrefix": true,
  "isModule": false,
  "isPublishedPackage": false,
  "isQuiet": false,
  "isUsingC8": false,
  "isUsingDependencyCruiser": false,
  "isUsingEslint": false,
  "isUsingNyc": false,
  "isUsingShuutils": false,
  "isUsingTailwind": false,
  "isUsingTypescript": true,
  "isUsingV8": false,
  "isUsingVue": false,
  "isWebPublished": false,
  "license": "MIT",
  "maxSizeKo": 1,
  "packageName": "",
  "repoId": "a-great-repo",
  "shouldAvoidSass": true,
  "userId": "Shuunen",
  "userIdLowercase": "shuunen",
  "userMail": "romain.racamier@gmail.com",
  "userName": "Michael Scott-Junior",
  "webUrl": "https://my-website.com",
  "willGenerateReport": true,
}
`;

exports[`augmentDataWithPackageJson D testFolder 1`] = `
ProjectData {
  "canAutoMergeDeps": true,
  "hasMainBranch": false,
  "hasTaskPrefix": false,
  "isModule": false,
  "isPublishedPackage": false,
  "isQuiet": false,
  "isUsingC8": false,
  "isUsingDependencyCruiser": false,
  "isUsingEslint": false,
  "isUsingNyc": false,
  "isUsingShuutils": false,
  "isUsingTailwind": false,
  "isUsingTypescript": false,
  "isUsingV8": false,
  "isUsingVue": false,
  "isWebPublished": false,
  "license": "GPL-3.0",
  "maxSizeKo": 1,
  "packageName": "",
  "repoId": "a-great-repo",
  "shouldAvoidSass": true,
  "userId": "Shuunen",
  "userIdLowercase": "shuunen",
  "userMail": "romain.racamier@gmail.com",
  "userName": "Romain Racamier-Lafon",
  "webUrl": "https://my-website.com",
  "willGenerateReport": true,
}
`;

exports[`objectToJson A nothing to sort 1`] = `
"{
  "keyA": "A"
}"
`;

exports[`objectToJson B already sorted 1`] = `
"{
  "keyA": "A",
  "keyB": "B"
}"
`;

exports[`objectToJson C sort 1`] = `
"{
  "keyA": "A",
  "keyB": "B"
}"
`;
